### PR TITLE
feat: replace worth-it scoring with boolean signal hybrid

### LINKED ISSUE #260: feat: Replace worth-it scoring with boolean signal hybrid
Part of #258

## Context

The current worth-it scoring uses 3 numeric dimensions (uniqueness 1-10, actionability 1-10, timeValue 1-10) averaged into a single score. This approach has two fundamental problems:

1. **LLMs can't calibrate absolute numeric ratings.** They either anchor high (everything 7+) or low (nothing above 6) depending on prompt pressure. There's no stable middle ground.
2. **"Uniqueness" is unjudgeable in isolation.** The LLM has no knowledge of what other podcasts cover, making this the noisiest dimension — it drags scores in unpredictable directions.

This sub-issue replaces the 3-dimension approach with a boolean signal hybrid (Option D from design discussion): 8 yes/no questions the LLM can reliably answer, producing a base score, plus a ±1 LLM adjustment for edge cases.

## Complexity Hint

**Subsystems touched:** Summarization prompt, ai-summary validation, SummaryResult type, score-utils, summary-display UI, database persistence
**Independent stories:** 1
**Structural changes needed:** Yes — `SummaryResult` type changes from `worthItDimensions: {uniqueness, actionability, timeValue}` to `worthItSignals` with boolean signals + adjustment
**Scope note:** Existing scored episodes will retain their old dimension data in the JSON column; new episodes get the new signal format

## Task

Replace the 3-dimension scoring prompt and validation with a boolean signal hybrid scoring system.

**Boolean signals (8 questions, each true/false):**
1. `hasActionableInsights` — Contains 3+ actionable insights
2. `hasNearTermApplicability` — Listener could apply something within a week
3. `staysFocused` — Episode stays focused, low filler-to-content ratio
4. `goesBeyondSurface` — Goes beyond surface-level discussion
5. `isWellStructured` — Well-structured and easy to follow
6. `timeJustified` — Time investment justified by content density
7. `hasConcreteExamples` — Includes concrete examples, data, or evidence
8. `hasExpertPerspectives` — Features expert/practitioner perspectives (not just opinions)

**Scoring formula:**
- Base score = count of `true` signals (0-8), mapped to 1.0-9.0 scale: `base = 1 + (trueCount * 1.0)`
- LLM adjustment = -1, 0, or +1 with mandatory justification
- Final score = clamp(base + adjustment, 1.0, 10.0)
- A score of 10 remains virtually impossible (requires all 8 signals + positive adjustment)

**Storage:** The `worthItDimensions` JSON column stores the new shape. Old episodes keep their `{uniqueness, actionability, timeValue}` data — the TypeScript type becomes a discriminated union.

## Acceptance Criteria

- [ ] VERIFY: Summarization prompt asks for 8 boolean signals instead of 3 numeric dimensions
- [ ] VERIFY: Prompt includes signal definitions so the LLM knows what each signal means
- [ ] VERIFY: `SummaryResult` type supports both old dimension format and new signal format
- [ ] VERIFY: `generateEpisodeSummary()` computes the score server-side from signals (doesn't trust LLM's arithmetic)
- [ ] VERIFY: Score formula: `base = 1 + trueCount`, then `clamp(base + adjustment, 1, 10)`
- [ ] VERIFY: LLM adjustment is constrained to exactly -1, 0, or +1 (reject other values, default to 0)
- [ ] VERIFY: `worthItReason` now includes which signals fired and the adjustment justification
- [ ] VERIFY: Summary display component shows which signals fired (8 signal checklist) instead of 3 dimension bars
- [ ] VERIFY: Old episodes with dimension-based scores still render correctly in the UI
- [ ] VERIFY: Score color thresholds in `score-utils.ts` remain appropriate for the new distribution
- [ ] VERIFY: `SCORE_THRESHOLD` in `dashboard.ts` is reviewed and adjusted if needed for the new distribution
- [ ] VERIFY: Unit tests cover: signal parsing, score computation, adjustment clamping, backward compat with old format
- [ ] VERIFY: `bun run lint`, `bun run test`, and `bun run build` pass

## Edge Cases

- LLM returns non-boolean values for signals — coerce truthy/falsy, log warning
- LLM returns adjustment outside [-1, 0, 1] — clamp to nearest valid value
- LLM omits adjustment or justification — default to 0 adjustment
- Old episodes with `{uniqueness, actionability, timeValue}` — UI detects old format and renders dimension bars as before
- Custom prompts (via admin `aiConfig`) — if present, skip signal-based scoring entirely (custom prompts own their own format)

### Files to Modify

- `src/lib/prompts.ts` — rewrite scoring section of `getSummarizationPrompt()` and `SYSTEM_PROMPT`: replace 3 dimensions with 8 boolean signals, add signal definitions, update JSON format example, update anti-inflation guide to signal-based calibration
- `src/lib/openrouter.ts` — update `SummaryResult` interface: add `worthItSignals` type, make `worthItDimensions` a union of old and new formats
- `src/trigger/helpers/ai-summary.ts` — rewrite score computation: count true signals → base score → apply adjustment → clamp. Replace the dimension averaging logic.
- `src/trigger/helpers/database.ts` — update `persistEpisodeSummary()` to store new signal format in `worthItDimensions` column
- `src/components/episodes/summary-display.tsx` — update score breakdown section: detect old vs new format, render signal checklist for new format, keep dimension bars for old format
- `src/lib/score-utils.ts` — review and adjust `getScoreColor()` and `getScoreLabel()` thresholds if needed for the new distribution
- `src/app/actions/dashboard.ts` — review `SCORE_THRESHOLD` constant for new score distribution
- `src/trigger/helpers/__tests__/ai-summary.test.ts` — rewrite score computation tests
- `src/trigger/helpers/__tests__/database.test.ts` — update persistence tests for new format

### Approach

1. Define the new `WorthItSignals` type and update `SummaryResult` in `src/lib/openrouter.ts`. Use a discriminated union so old dimension data and new signal data coexist.
2. Rewrite the scoring sections in `src/lib/prompts.ts`: replace dimension definitions with signal definitions, update the JSON format example, revise calibration guidance.
3. Rewrite score computation in `src/trigger/helpers/ai-summary.ts`: parse boolean signals, count trues, compute base, validate and apply adjustment, clamp.
4. Update `persistEpisodeSummary()` in `src/trigger/helpers/database.ts` to persist the new signal structure.
5. Update `summary-display.tsx` to render a signal checklist (green check / red x for each signal) instead of dimension progress bars. Add format detection to fall back to old rendering for existing episodes.
6. Review `score-utils.ts` thresholds and `SCORE_THRESHOLD` in `dashboard.ts` — adjust if the new score distribution shifts significantly.
7. Rewrite unit tests for the new scoring logic.
8. Test end-to-end: summarize an episode, verify score breakdown shows signals.

### Patterns to Follow

- Follow the existing validation pattern in `ai-summary.ts` lines 40-54 for server-side score recomputation.
- Follow the existing `SummaryResult` interface pattern for backward-compatible type evolution.
- Match the existing summary-display.tsx section layout for the signal checklist rendering.

### Constraints

- **Backward compatibility:** Old episodes with `{uniqueness, actionability, timeValue}` must continue to display correctly. No data migration for existing episodes.
- **Custom prompts:** If `aiConfig.summarizationPrompt` is set, the custom prompt owns its scoring format. Signal-based scoring only applies to the default prompt.
- **Score range:** Must stay 1.0-10.0 to maintain compatibility with existing `worth_it_score` column constraints and UI color thresholds.

### In Scope
- Rewriting the scoring prompt from dimensions to boolean signals
- Server-side score computation from signals
- UI update to show signal breakdown
- Backward compatibility with old dimension scores
- Unit tests

### Out of Scope
- Backfilling old episodes with new signal-based scores
- Cross-episode topic ranking (sub-issue #3)
- Personal score adjustment (sub-issue #4)
- Changing the `worth_it_score` database column type or constraints

## Dependencies

- Blocked by: None (independent of sub-issue #1)
- Blocks: None directly, but improves the quality of Layer 1 scores that feed into sub-issues #3 and #4

## Testing Requirements

- [ ] Unit tests for signal parsing and normalization (truthy/falsy coercion)
- [ ] Unit tests for score computation (0 signals through 8 signals, each adjustment value)
- [ ] Unit tests for adjustment clamping (values outside [-1, 0, 1])
- [ ] Unit tests for backward compat: old `SummaryResult` with dimensions still processes correctly
- [ ] Component test for summary-display: renders signals for new format, dimensions for old format

## Definition of Done

- [ ] Implementation complete
- [ ] Tests passing (unit + integration)
- [ ] `bun run lint`, `bun run test`, `bun run build` pass
- [ ] PR approved and merged
- [ ] Epic checklist updated

### PR DESCRIPTION
**Pull request by funny-otter**
## Summary

- Replace 3-dimension numeric scoring (uniqueness/actionability/timeValue averaged) with 8 boolean signal questions + constrained ±1 LLM adjustment for more stable, interpretable episode scores
- Score formula: `clamp(1 + count_of_true_signals + adjustment, 1, 10)` — LLMs answer yes/no reliably vs struggling with 1-10 calibration
- Backward compatible: old episodes retain dimension-based scores with legacy progress bars; new episodes show a signal checklist UI
- No database migration needed — the JSON column accepts both formats via a discriminated union (`kind: "signals"` vs legacy)
- Existing score thresholds (8/6/4/2 display tiers, 6.00 recommendations) remain semantically valid for the new distribution
- ADR-032 documents the scoring model change, design decisions, and threshold analysis

### Files changed (16 files, +625/-79)
- **Types:** `openrouter.ts` — `WorthItSignals`, `WorthItDimensionsData` discriminated union, `SIGNAL_LABELS`
- **Helpers:** `score-utils.ts` — `computeSignalScore()`, `clampAdjustment()`, `coerceSignals()`
- **Prompts:** `prompts.ts` — 8 boolean signal questions + adjustment instructions
- **Scoring:** `ai-summary.ts` — 3-tier fallback: signals → dimensions → raw score; `console.warn` for non-boolean coercion
- **Schema:** `schema.ts` — updated `$type<>` annotation (no migration)
- **UI:** `summary-display.tsx` — signal checklist with CheckCircle/XCircle icons; legacy progress bars preserved
- **Tests:** 30 score-utils tests, signal computation/coercion/clamping tests, updated database + prompt fixtures
- **Docs:** ADR-032, CHANGELOG

## Test plan

- [x] `bun run lint` — clean
- [x] `bun run test` — 1540 tests pass (129 files)
- [x] `bun run build` — production build succeeds
- [ ] Storybook: verify `WithSignals` and `WithLegacyDimensions` stories render correctly
- [ ] E2E: summarize a real episode and verify signal checklist appears in the UI
- [ ] Verify old episodes with dimension-based scores still render with legacy progress bars

Closes #260

## Summary by CodeRabbit

* **Changed**
  * Episode scoring now derives from eight boolean quality signals plus a ±1 adjustment, producing more stable, interpretable episode scores while preserving legacy compatibility.
  * Episode pages now display per-signal pass/fail indicators and an optional adjustment line; the score display supports both new and legacy formats.
* **Documentation**
  * Changelog and design docs updated to describe the signal-based scoring and transition behavior.

### CODE REVIEW

File: src/trigger/helpers/ai-summary.ts [hunk 1]
```diff
@@ -40,24 +42,73 @@ export async function generateEpisodeSummary(
...
+    const rawSignals = raw.worthItSignals;
+    if (rawSignals && typeof rawSignals === "object") {
+      const signalObj = rawSignals as Record<string, unknown>;
+      // Warn on non-boolean signal values before coercion
+      for (const key of WORTH_IT_SIGNAL_KEYS) {
+        if (typeof signalObj[key] !== "boolean") {
+          console.warn(
+            `[ai-summary] non-boolean signal coerced: ${key}=${signalObj[key]} → ${Boolean(signalObj[key])}`
+          );
+        }
```
[the remark this answers is not in the document]
Comment (funny-otter) [reply src/trigger/helpers/ai-summary.ts]: Fixed: added `key in signalObj` guard so missing keys no longer trigger warnings — only present non-boolean values are logged.

File: src/trigger/helpers/ai-summary.ts [hunk 2]
```diff
@@ -40,24 +42,73 @@ export async function generateEpisodeSummary(
...
+      const firedLabels = WORTH_IT_SIGNAL_KEYS
+        .filter((k) => signals[k])
+        .map((k) => SIGNAL_LABELS[k].toLowerCase());
+      const firedCount = firedLabels.length;
+      const adjText =
+        adjustment === 0
+          ? `Adjustment: 0 (${adjustmentReason || "signals capture quality accurately"}).`
+          : `Adjustment: ${adjustment > 0 ? "+1" : "-1"} (${adjustmentReason}).`;
+      result.worthItReason =
+        `${firedCount}/8 signals: ${firedLabels.join(", ") || "none"}. ${adjText}`;
```
[the remark this answers is not in the document]
Comment (funny-otter) [reply src/trigger/helpers/ai-summary.ts]: Fixed: worthItReason now preserves the LLM's qualitative verdict and appends the signal summary in parentheses. Falls back to signal-only format when no LLM reason is present.

File: src/trigger/helpers/ai-summary.ts [hunk 3]
```diff
@@ -40,24 +42,73 @@ export async function generateEpisodeSummary(
...
         result.worthItScore = computed;
+        result.worthItDimensions = {
+          kind: "dimensions",
+          uniqueness,
+          actionability,
+          timeValue,
+        };
       }
     }
+    // Neither format (custom prompt returning raw score) — keep as-is
```
[the remark this answers is not in the document]
Comment (funny-otter) [reply src/trigger/helpers/ai-summary.ts]: Fixed: added default `worthItScore = 5` when neither signals, dimensions, nor raw LLM score is present. Added test coverage.

File: src/components/episodes/summary-display.stories.tsx [hunk 4]
```diff
@@ -70,3 +70,46 @@ export const LowScore: Story = {
...
+        staysFocused: true,
+        goesBeyondSurface: true,
+        isWellStructured: true,
+        timeJustified: false,
+        hasConcreteExamples: false,
+        hasExpertPerspectives: false,
+      },
+      adjustment: 1,
+      adjustmentReason:
+        "Exceptional guest lineup elevates the content beyond the checklist.",
```
[the remark this answers is not in the document]
Comment (funny-otter) [reply src/components/episodes/summary-display.stories.tsx:100]: Fixed: updated worthItReason to list "well-structured" instead of "concrete examples" to match the signal booleans (isWellStructured: true, hasConcreteExamples: false).

File: src/db/schema.ts [hunk 5]
```diff
@@ -88,11 +89,10 @@ export const episodes = pgTable(
...
     worthItReason: text("worth_it_reason"),
-    worthItDimensions: json("worth_it_dimensions").$type<{
-      uniqueness: number;
-      actionability: number;
-      timeValue: number;
-    }>(),
+    worthItDimensions: json("worth_it_dimensions").$type<
+      | { kind: "signals"; signals: WorthItSignals; adjustment: -1 | 0 | 1; adjustmentReason: string }
+      | { uniqueness: number; actionability: number; timeValue: number }
+    >(),
```
[the remark this answers is not in the document]
Comment (funny-otter) [reply src/db/schema.ts:95]: Fixed: added `kind: "dimensions"` to the legacy variant in the schema `$type<>()` annotation, aligning it with the `WorthItDimensionsData` discriminated union.

File: src/lib/score-utils.ts [hunk 6]
```diff
@@ -18,3 +21,33 @@ export function getScoreLabel(score: number): string {
...
+  return {
+    hasActionableInsights: Boolean(raw.hasActionableInsights),
+    hasNearTermApplicability: Boolean(raw.hasNearTermApplicability),
+    staysFocused: Boolean(raw.staysFocused),
+    goesBeyondSurface: Boolean(raw.goesBeyondSurface),
+    isWellStructured: Boolean(raw.isWellStructured),
+    timeJustified: Boolean(raw.timeJustified),
+    hasConcreteExamples: Boolean(raw.hasConcreteExamples),
+    hasExpertPerspectives: Boolean(raw.hasExpertPerspectives),
+  };
```
[the remark this answers is not in the document]
Comment (funny-otter) [reply src/lib/score-utils.ts:56]: Fixed: replaced `Boolean()` with `toSignalBoolean()` that explicitly handles string "true"/"false" (case-insensitive), numeric 0/1, and defaults unknown values to false. Added regression test for string "false".

File: src/trigger/helpers/__tests__/database.test.ts [hunk 7]
```diff
@@ -135,9 +135,23 @@ describe("persistEpisodeSummary", () => {
...
+        staysFocused: true,
+        goesBeyondSurface: true,
+        isWellStructured: true,
+        timeJustified: true,
+        hasConcreteExamples: true,
+        hasExpertPerspectives: false,
+      },
+      adjustment: 0 as const,
+      adjustmentReason: "Signals capture quality accurately.",
+    },
```
[the remark this answers is not in the document]
Comment (funny-otter) [reply src/trigger/helpers/__tests__/database.test.ts]: Fixed: updated fixture worthItReason from "5/8" to "6/8" and added "time justified" to the signal label list to match the 6 true signals in the data.

**Comment by funny-otter** [comment]
> Actionable comments posted: 4...

Fixed: added `text` language specifier to the ADR code fence as suggested in the nitpick. All 4 actionable inline comments have been addressed and resolved.

**Comment by funny-otter** [comment]
> Code Review — This pull request implements a new "Boolean Signal Hybrid" scoring system...

Fixed: all three inline suggestions addressed — added `key in` guard for warn logging, preserved LLM verdict in worthItReason, and added default score fallback.
